### PR TITLE
src: use un-deprecated version of aio_create_completion

### PR DESCRIPTION
**Pull request by adamemerson**
Clean up various warnings under GCC and Clang.

### CODE REVIEW

File: src/crush/CrushWrapper.cc [hunk 1]
```diff
@@ -932,7 +932,7 @@ int CrushWrapper::verify_upmap(CephContext *cct,
     return -ENOENT;
   }
   int root_bucket = 0;
-  int cursor = 0;
+  unsigned cursor = 0;
```
Comment (tchaikov) [inline src/crush/CrushWrapper.cc]: could you drop this change, as it's already in #31184

Comment (adamemerson) [reply src/crush/CrushWrapper.cc]: Done!

**Comment by tchaikov** [comment]
retest this please.

```
[ RUN      ] TestMockObjectMapLockRequest.GetLockInfoMissing
[    Errors while running CTest
Build step 'Execute shell' marked build as failure
[PostBuildScript] - Executing post build scripts.
[ceph-pull-requests] $ /bin/sh -xe /tmp/jenkins9194797019727076001.sh
+ sudo reboot
```

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2019-11-11_03:49:19-rados-wip-kefu2-testing-2019-11-03-0500-distro-basic-mira/

**Pull request merged**